### ISSUE TITLE
Able to set up filters, but they are not filtering on the page

### ISSUE THREAD

**Issue by mwentz** [Jetski5822/Orchard.Taxonomies.ProjectionFilters#1]
These filters are exactly what I need (and I think they should probably be added to the Orchard Core), but I have not been able to get them to work. I tried adding the four filter files to the Orchard.Taxonomies/Projections folder and I was able to compile the site and set up the filter in a query, but the filtering didn't seem to be working. I also tried to add this as a separate module (see https://orchard.codeplex.com/workitem/21214) and had the same issue.

**Comment by Jetski5822** [comment]
@ultimateawesome these filters do two things.
1. One filter works if you have term={id} in the url. That is the URL Filter.
2. One filter will filter by related Terms. so....

If Page A has term Foo from taxonomy TBar
If Page B has term Bar from taxonomy TBar
If Page C has term Foo from taxonomy TBar

If you applied the related terms filter, and then navigated to Page C, you would see Page A and C in the filter.

If you then navigated to B, you would only see B.

Does that help?

**Comment by mwentz** [comment]
Thanks Nick, that is very helpful. I am using the Related Terms filter and it is now working for me.

The reason I was not able to get it to work before was because I was using custom content types instead of pages. If I create a page and add a query widget using related terms, the query filters properly. If I create a custom content type and add a query widget using related terms, the query does not filter properly.

It seems that the term is not being identified properly by the filter for non-page content types. If I add a query widget on the page for a custom content item called "related pages" with Content Type: Page and the related terms filter selected, it shows all the pages unfiltered. (If I just add a filter on that custom content item that doesn't include a content type filter I get an out of memory exception, a related issue but I will open a new bug for that).

Does that all make sense? I'm happy to try to come up with an example if you need.

**Comment by mwentz** [comment]
I figured out what the issue was. Not sure why this is, but this filter didn't seem to work until I added an Autoroute part to the content item with a custom path.

I created a custom content type Success Story with Title, Body, and Taxonomy (TBar) parts.

Success Story A has term Foo from taxonomy TBar
Success Story B has term Bar from taxonomy TBar
Success Story C has term Foo from taxonomy TBar

I applied the related terms filter and navigating to Success Story A, B and C all display all three pages.

I added the Autoroute part to Success Story and added {Content.ContentType.Slug}/{Content.DisplayText.Slug} as the pattern. I republished all three success stories with the new pattern applied. Once I did that, the related terms filter works as expected.

This solves my problem for now since I will use a custom autoroute pattern for all my pages. I think that the filter should work even without a custom autoroute, though.
